iree/hal/cuda/registration/driver_module.cc

```diff
@@ -0,0 +1,69 @@
+// Copyright 2021 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "iree/hal/cuda/registration/driver_module.h"
+
+#include <inttypes.h>
+
+#include "iree/base/status.h"
+#include "iree/base/target_platform.h"
+#include "iree/base/tracing.h"
+#include "iree/hal/cuda/api.h"
+
+#define IREE_HAL_CUDA_DRIVER_ID 0x43554441u  // CUDA
+
+static iree_status_t iree_hal_cuda_driver_factory_enumerate(
+    void* self, const iree_hal_driver_info_t** out_driver_infos,
+    iree_host_size_t* out_driver_info_count) {
+  // NOTE: we could query supported cuda versions or featuresets here.
+  static const iree_hal_driver_info_t driver_infos[1] = {{
+      /*driver_id=*/IREE_HAL_CUDA_DRIVER_ID,
```

Comment:
when this file is a .c these can become the named initializers too (`.driver_id = IREE_HAL_CUDA_DRIVER_ID`)

For files that don't need C++ - like this one - I'd recommend making them .c right now. It will help catch a lot of issues that compiling the files as C++ masks.